### DISCUSSION TITLE
Option to ignore CSS when organizing imports

### DISCUSSION THREAD

**Discussion by remojansen** [biomejs/biome#807] [Ideas]
Hi, I'm trying to migrate a very large project (@ematipico knows this project) to Biome, and I'm having an issue preventing the dev team from adopting Biome.

The issue is the order of CSS imports. Biome is sorting the imports, but this breaks the web app. We are aware of the possibility of adding line breaks to prevent this issue. However, the dev team feels this opens the door for human errors.

We would like an option to organise imports (just the way it is now) but avoid changing the order of CSS files. Something like:

```json
"organizeImports": {
    "enabled": true,
    "organizeCSS": false
}
```

We don't mind if all CSS files are placed at the top, but the order within the files should not be altered. By doing this, we will not have to worry about one of our developers forgetting to add a line break between CSS files.

Please let us know your thoughts. Thanks!

## Update  (21 Nov 2023)

Confirmed by the Biome team there is no need for new config. The issues should be addressed by a bug fix.

**Comment by Conaclos** [comment]
Hi! Thanks for reporting this issue!

Could you provide a concrete example with an example on the Biome's playground? This could help to picture the issue.

**Comment by remojansen** [comment]
Sure!

Here it is:

[Playground demo](https://biomejs.dev/playground/?bracketSpacing=false&code=aQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwBuAG8AcgBtAGEAbABpAHoAZQAuAGMAcwBzACcAOwAKAGkAbQBwAG8AcgB0ACAAJwAuAC8AaQBuAGQAZQB4AC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwBiAHUAdAB0AG8AbgAuAGMAcwBzACcAOwAKAGkAbQBwAG8AcgB0ACAAJwBAAHAAcgBvAGoAZQBjAHQAeAAvAHQAaABlAG0AZQAvAHMAcgBjAC8AYwBvAGwALQAzAC0AcwBlAGMAdABpAG8AbgAuAGMAcwBzACcAOwAKAGkAbQBwAG8AcgB0ACAAJwBAAHAAcgBvAGoAZQBjAHQAeAAvAHQAaABlAG0AZQAvAHMAcgBjAC8AYwBvAGwALQA1ADAAeAA1ADAALQBzAGUAYwB0AGkAbwBuAC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwBjAG8AbAAtAGYAdQBsAGwALQBzAGUAYwB0AGkAbwBuAC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwBvAG4AZQB0AHIAdQBzAHQALgBjAHMAcwAnADsACgBpAG0AcABvAHIAdAAgACcAQABwAHIAbwBqAGUAYwB0AHgALwB0AGgAZQBtAGUALwBzAHIAYwAvAHIAaQBjAGgALQB0AGUAeAB0AC0AYQBjAGMAbwByAGQAaQBvAG4ALgBjAHMAcwAnADsACgBpAG0AcABvAHIAdAAgACcAQABwAHIAbwBqAGUAYwB0AHgALwB0AGgAZQBtAGUALwBzAHIAYwAvAHIAaQBjAGgALQB0AGUAeAB0AC0AdABhAGIAbABlAC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwBzAGkAZABlAC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwBzAHUAbQBtAGEAcgB5AC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AdABoAGUAbQBlAC8AcwByAGMALwB1AHQAaQBsAHMALgBjAHMAcwAnADsACgBpAG0AcABvAHIAdAAgACcAQABwAHIAbwBqAGUAYwB0AHgALwB0AGgAZQBtAGUALwBzAHIAYwAvAGMAbwBtAHAAbwBuAGUAbgB0AC0AdQBpAC0AZgBvAHIAbQAuAGMAcwBzACcAOwAKAGkAbQBwAG8AcgB0ACAAJwBAAHAAcgBvAGoAZQBjAHQAeAAvAHQAaABlAG0AZQAvAHMAcgBjAC8AbQBhAGkAbgAtAGEAcABwAC4AYwBzAHMAJwA7AAoAaQBtAHAAbwByAHQAIAB7ACAATABhAHkAbwB1AHQASQBuAHQAZQByAG4AYQBsACAAfQAgAGYAcgBvAG0AIAAnAC4ALwBMAGEAeQBvAHUAdABJAG4AdABlAHIAbgBhAGwAJwA7AAoAaQBtAHAAbwByAHQAIAB7ACAATABvAGcAaQBuAFMAdABhAHQAdQBzACAAfQAgAGYAcgBvAG0AIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AYwBvAG4AcwB0AGEAbgB0AHMALQBjAG8AbQBtAG8AbgAnADsACgBpAG0AcABvAHIAdAAgAHsAIABFAHIAcgBvAHIASABhAG4AZABsAGkAbgBnACAAfQAgAGYAcgBvAG0AIAAnAEAAcAByAG8AagBlAGMAdAB4AC8AYwBvAG4AdABhAGkAbgBlAHIALQBlAHIAcgBvAHIALQBoAGEAbgBkAGwAZQByACcAOwAKAGkAbQBwAG8AcgB0ACAAewAgAEMAdQBzAHQAbwBtAEUAcgByAG8AcgAgAH0AIABmAHIAbwBtACAAJwAuAC8AQwB1AHMAdABvAG0ARQByAHIAbwByACcAOwAKAGkAbQBwAG8AcgB0ACAAUgBlAGEAYwB0ACAAZgByAG8AbQAgACcAcgBlAGEAYwB0ACcAOwA%3D)

Input:

```ts
import '@projectx/theme/src/normalize.css';
import './index.css';
import '@projectx/theme/src/button.css';
import '@projectx/theme/src/col-3-section.css';
import '@projectx/theme/src/col-50x50-section.css';
import '@projectx/theme/src/col-full-section.css';
import '@projectx/theme/src/onetrust.css';
import '@projectx/theme/src/rich-text-accordion.css';
import '@projectx/theme/src/rich-text-table.css';
import '@projectx/theme/src/side.css';
import '@projectx/theme/src/summary.css';
import '@projectx/theme/src/utils.css';
import '@projectx/theme/src/component-ui-form.css';
import '@projectx/theme/src/main-app.css';
import { LayoutInternal } from './LayoutInternal';
import { LoginStatus } from '@projectx/constants-common';
import { ErrorHandling } from '@projectx/container-error-handler';
import { CustomError } from './CustomError';
import React from 'react';
```

Output (Import sorting):
```ts
import { LoginStatus } from '@projectx/constants-common';
import { ErrorHandling } from '@projectx/container-error-handler';
import '@projectx/theme/src/button.css';
import '@projectx/theme/src/col-3-section.css';
import '@projectx/theme/src/col-50x50-section.css';
import '@projectx/theme/src/col-full-section.css';
import '@projectx/theme/src/component-ui-form.css';
import '@projectx/theme/src/main-app.css';
import '@projectx/theme/src/normalize.css';
import '@projectx/theme/src/onetrust.css';
import '@projectx/theme/src/rich-text-accordion.css';
import '@projectx/theme/src/rich-text-table.css';
import '@projectx/theme/src/side.css';
import '@projectx/theme/src/summary.css';
import '@projectx/theme/src/utils.css';
import React from 'react';
import { CustomError } from './CustomError';
import { LayoutInternal } from './LayoutInternal';
import './index.css';
```

What we would like (`organizeCSS` `false`):
```ts
import '@projectx/theme/src/normalize.css';
import './index.css';
import '@projectx/theme/src/button.css';
import '@projectx/theme/src/col-3-section.css';
import '@projectx/theme/src/col-50x50-section.css';
import '@projectx/theme/src/col-full-section.css';
import '@projectx/theme/src/onetrust.css';
import '@projectx/theme/src/rich-text-accordion.css';
import '@projectx/theme/src/rich-text-table.css';
import '@projectx/theme/src/side.css';
import '@projectx/theme/src/summary.css';
import '@projectx/theme/src/utils.css';
import '@projectx/theme/src/component-ui-form.css';
import '@projectx/theme/src/main-app.css';
import { LoginStatus } from '@projectx/constants-common';
import { ErrorHandling } from '@projectx/container-error-handler';
import React from 'react';
import { CustomError } from './CustomError';
import { LayoutInternal } from './LayoutInternal';
```

Thanks

**Reply by Conaclos** [reply]
Biome should indeed not reorder these imports. It should consider them as side-effect imports.

**Reply by remojansen** [reply]
So, is the current behaviour not expected? Should a bug be raised, or will a new option/feature be required?

**Reply by ematipico** [reply]
Yeah it's definitely a bug!

**Reply by remojansen** [reply]
I am happy to try to work on a bug fix over the Christmas break. It will be my first time working with Rust and contributing to Biome. I will probably need some help, but I'm feeling brave. I wanted to give Rust a go, so this could be my perfect excuse.

**Reply by ematipico** [reply]
Feel free to open an issue, so we can assign that to you. You can use that issue to ask any questions you need 👌

**Comment by remojansen** [comment]
The bug is https://github.com/biomejs/biome/issues/817 I will now close this discussion
